### PR TITLE
Fix field mapping check

### LINKED ISSUE #689: collections/_import endpoint crashes if no index is found
even if an index template has been uploaded. It must not crash if a template matches `collection.indexName` pattern.

```
! org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=index_not_found_exception, reason=no such index [city_passages]]
! at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:187)
! at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1892)
! at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1869)
! at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1626)
! at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1598)
! at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1565)
! at org.elasticsearch.client.IndicesClient.getFieldMapping(IndicesClient.java:553)
! at io.arlas.server.impl.elastic.utils.ElasticClient.getFieldMapping(ElasticClient.java:185)
! at io.arlas.server.impl.elastic.utils.ElasticTool.checkIndexMappingFields(ElasticTool.java:142)
! at io.arlas.server.impl.elastic.utils.ElasticTool.checkAliasMappingFields(ElasticTool.java:156)
! at io.arlas.server.impl.elastic.dao.ElasticCollectionReferenceDao.checkCollectionReferenceParameters(ElasticCollectionReferenceDao.java:206)
! at io.arlas.server.impl.elastic.dao.ElasticCollectionReferenceDao.putCollectionReference(ElasticCollectionReferenceDao.java:218)
! at io.arlas.server.rest.collections.CollectionService.save(CollectionService.java:257)
```

This can be a regression following changes in template management in Elasticsearch.

### PR DESCRIPTION
**Pull request by alainbodiguel**
closes #689

### CODE REVIEW

File: arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java [hunk 1]
```diff
@@ -136,38 +136,36 @@ public static AcknowledgedResponse putExtendedMapping(ElasticClient client,
...
-        List<String> indices = ElasticTool.getIndicesName(client, alias);
-        for (String index : indices) { checkIndexMappingFields(client, index, fields); }
-        return true;
-    }
-
-    public static List<String> getIndicesName(ElasticClient client, String alias) throws ArlasException {
+    public static Map<String, LinkedHashMap> checkAliasMappingFields(ElasticClient client,
+                                                                     String alias,
+                                                                     String... fields) throws ArlasException {
         Map<String, LinkedHashMap> response = client.getMappings(alias);
```
Comment (MohamedHamouGisaia) [inline arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java:159]: I realise that we can call the `cacheManager.getMapping(alias)` method to exploit the cache at this level

File: arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java [hunk 2]
```diff
@@ -136,38 +136,36 @@ public static AcknowledgedResponse putExtendedMapping(ElasticClient client,
...
 
         List<String> indices = IteratorUtils.toList(response.keySet().iterator());
         for (String index : indices) {
-            Object properties = response.get(index);
-            if (properties == null) {
-                throw new NotFoundException("Unable to find properties in " + index + ".");
+            LinkedHashMap properties = response.get(index);
```
Comment (MohamedHamouGisaia) [inline arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java:163]: We still have the risk of `properties` being null no ?

File: arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java (hunk 1 above)
Comment (alainbodiguel) [reply arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java:159]: Good point. I've done it.

File: arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java (hunk 2 above)
Comment (alainbodiguel) [reply arlas-core/src/main/java/io/arlas/server/impl/elastic/utils/ElasticTool.java:163]: Good point. Fixed.

**Review by MohamedHamouGisaia — APPROVED** [review]

**Review by mbarbet — APPROVED** [review]

**Pull request merged**